src/meltano/core/state_service.py

```diff
@@ -31,13 +31,16 @@ class StateService:
     Currently only manages Singer state for Extract and Load jobs.
     """
 
-    def __init__(self, session: object = None):
```

Comment:
Passing in the project object, and letting the `StateService` create the `ProjectSettingsService` itself rather than putting that responsibility on all of the callers would be preferable.

Reply:
I've tried this way at the beginning, but ProjectSettingsService leads to a circular dependency. Do you have another better approach we can try?